source/agora/network/NetworkManager.d

```diff
@@ -241,28 +241,46 @@ public class NetworkManager
 
         logInfo("Establishing connection with %s...", address);
         auto node = new NetworkClient(this.taskman, address,
-            this.getClient(address), this.node_config.retry_delay.msecs);
+            this.getClient(address), this.node_config.retry_delay.msecs,
+            this.node_config.max_retries);
 
-        node.handshake();
-        this.onNodeConnected(node);
+        while (1)
+        {
+            try
+            {
+                node.handshake();
+                this.onNodeConnected(node);
+                break;
+            }
+            catch (Exception ex)
+            {
+               // keep trying until node is connected
```

Comment:
Btw, this part is changed in the node banning support code. For now it has the same semantics as before, keep trying to handshake.

Reply:
As in, it's changed in one of the other PRs.